src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -769,9 +770,50 @@ HRESULT _stdcall TerminalResize(void* terminal, COORD dimensions)
     publicTerminal->_terminal->ClearSelection();
     publicTerminal->_renderer->TriggerRedrawAll();
 
+    // Resize the text buffer without changing the renderer draw size.
     return publicTerminal->_terminal->UserResize(dimensions);
 }
 
+/// <summary>
+/// Resizes the renderer to fit the provided width and height without changing the text
+/// buffer size.
+/// </summary>
+HRESULT _stdcall ResizeRendererDrawSpace(void* terminal, double width, double height, _Out_ COORD* dimensions)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, dimensions);
+
+    const auto publicTerminal = static_cast<HwndTerminal*>(terminal);
+
+    auto lock = publicTerminal->_terminal->LockForWriting();
+    publicTerminal->_terminal->ClearSelection();
+
+    LOG_IF_WIN32_BOOL_FALSE(SetWindowPos(
+        publicTerminal->GetHwnd(),
+        nullptr,
+        0,
+        0,
+        static_cast<int>(width),
+        static_cast<int>(height),
+        0));
+
+    const SIZE windowSize{ static_cast<short>(width), static_cast<short>(height) };

```

Comment:
If we're going to shrink down the width/height to `short` anyway, maybe the method signature should just take a `short` in the first place and make it the caller's problem as to what to do when the value won't fit?

We don't necessarily want to set ourselves up for the circumstance of losing the precision in different ways (`SetWindowPos` to `int` and then again here to `short`.)